### PR TITLE
Added AADConnectorGroupApplicationProxy resource

### PR DESCRIPTION
**Pull request by ritikmit**
#### Pull Request (PR) description
Added AADConnectorGroupApplicationProxy resource

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADConnectorGroupApplicationProxy/MSFT_AADConnectorGroupApplicationProxy.schema.mof [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+[ClassVersion("1.0.0.0"), FriendlyName("AADConnectorGroupApplicationProxy")]
+class MSFT_AADConnectorGroupApplicationProxy : OMI_BaseResource
+{
+    [Write, Description("The name associated with the connectorGroup.")] String Name;
+    [Write, Description("The region the connectorGroup is assigned to and will optimize traffic for. This region can only be set if no connectors or applications are assigned to the connectorGroup. The possible values are: nam (for North America), eur (for Europe), aus (for Australia), asia (for Asia), ind (for India), and unknownFutureValue."), ValueMap{"nam","eur","aus","asia","ind","unknownFutureValue"}, Values{"nam","eur","aus","asia","ind","unknownFutureValue"}] String Region;
+    [Key, Description("The unique identifier for an entity. Read-only.")] String Id;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADConnectorGroupApplicationProxy/MSFT_AADConnectorGroupApplicationProxy.schema.mof]: Again, we should probably make Name the Key property to ensure we can replicate across tenants.

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADConnectorGroupApplicationProxy/MSFT_AADConnectorGroupApplicationProxy.psm1 [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+        $Name,
+
+        [Parameter()]
+        [ValidateSet('nam','eur','aus','asia','ind','unknownFutureValue')]
+        [System.String]
+        $Region,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Id,
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADConnectorGroupApplicationProxy/MSFT_AADConnectorGroupApplicationProxy.psm1:18]: Should name be the Key to ensure we can replicate across tenants, or are ID immutable across tenants?

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
Comments to be addressed around the Key property

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADConnectorGroupApplicationProxy/MSFT_AADConnectorGroupApplicationProxy.psm1 (hunk 2 above)
Comment (ritikmit) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADConnectorGroupApplicationProxy/MSFT_AADConnectorGroupApplicationProxy.psm1:18]: Done

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**